spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -240,8 +240,12 @@ public String getBinder(String channelName) {
 	}
 
 	public BindingProperties getBindingProperties(String channelName) {
-		BindingProperties bindingProperties = bindings.containsKey(channelName) ?
-				bindings.get(channelName) : new BindingProperties();
+		if (this.bindings.containsKey(channelName)) {
+			return this.bindings.get(channelName);
+		}
+		// Update the bindings map with the new Binding properties.
+		BindingProperties bindingProperties = new BindingProperties();
```

Comment:
I don't think we should ever update the property map for bindings nor should we update the binding instance. What can we do instead is to return a copy that is populated with the default value (i.e. the value that is supposed to override the unset value).